Sources/MongoSwift/BSON/Document.swift

```diff
@@ -246,7 +246,7 @@ extension Document {
             return self
         }
 
-        var idDoc: Document = ["_id": .objectID(ObjectID())]
+        var idDoc: Document = ["_id": .objectID(BSONObjectID())]
```

Comment:
I don't mean to re-open a can of worms here, but this looks cumbersome to me. Was it not possible to have something like:
```
var idDoc: Document = ["_id": .objectID(1) | .objectID("hexvalue") | .objectID()]
```
?

I understand changing the enum requires breaking changes, but the inputs for creating an `ObjectID` are unlikely to change, and you could still accept an `ObjectID` instance for those special cases (if they exist).

Reply:
to address each example you've given - 
* `"_id": .objectID(1)`: 1 isn't a valid ObjectID, so I'm not sure what you would expect the behavior to be there?
* `.objectID("hexvalue")`: we could actually support this by making `ObjectID` conform to `ExpressibleByStringLiteral`. but we lose the ability to validate your input string since we can't throw from that constructor. this is kind of problematic because in the case of constructing a document via dictionary literal I think we'd have to just fatal error if your input is invalid and can't be encoded to BSON.
* `.objectID()`: since Swift 5.1 now allows you to provide default values for enum associated values, we can do this easily by updating the case from `.objectID(ObjectID)` to `.objectID(ObjectID = ObjectID())`.

I don't love the `ExpressibleByStringLiteral` change for the reason I mentioned about validation, but the last one (adding default value) I am definitely in favor of.

Reply:
cool, yes that default value approach is basically what I'm looking for!

Reply:
added. opened [SWIFT-870](https://jira.mongodb.org/browse/SWIFT-870) for changelog purposes